pkg/services/notary/notary.go

```diff
@@ -103,9 +107,38 @@ func NewNotary(bc blockchainer.Blockchainer, log *zap.Logger) (*Notary, error) {
 			Log:     log,
 		},
 		wallet: wallet,
+		mp:     mp,
+		reqCh:  make(chan mempool.Event),
+		stopCh: make(chan struct{}),
 	}, nil
 }
 
+// Run runs Notary module and should be called in a separate goroutine.
+func (n *Notary) Run() {
+	n.mp.SubscribeForTransactions(n.reqCh)
+	for {
+		select {
+		case <-n.stopCh:
+			n.mp.UnsubscribeFromTransactions(n.reqCh)
+			return
+		case event := <-n.reqCh:
+			if req, ok := event.Data.(*payload.P2PNotaryRequest); ok {
+				switch event.Type {
+				case mempool.TransactionAdded:
+					n.OnNewRequest(req)
+				case mempool.TransactionRemoved:
+					n.OnFallbackRemoval(req)
```

Comment:
`OnRequestRemoval`?

Reply:
We may call it `OnRequestRemoval`, but it may mislead the users of it because only fallback transaction is being removed (while the whole request may still be valid). Are you sure about it?

Reply:
Frankly, to me "fallback removal" is more confusing because I was wondering how main transactions are removed. And I'm for keeping some symmetry, if "request" is being added two lines above then it's "request" that should removed here.